source/framework/vx_user_data_object.c

```diff
@@ -177,6 +244,28 @@ VX_API_ENTRY vx_user_data_object VX_API_CALL vxCreateUserDataObject(
     return (user_data_object);
 }
 
+vx_user_data_object ownCreateReadOnlyUserDataObject(vx_user_data_object parent)
+{
+    vx_user_data_object user_data_object = NULL;
+    user_data_object = (vx_user_data_object)ownCreateReference(parent->base.context, (vx_enum)VX_TYPE_USER_DATA_OBJECT, (vx_enum)VX_EXTERNAL, &parent->base.context->base);
+
+    if ((vxGetStatus((vx_reference)user_data_object) == (vx_status)VX_SUCCESS) &&
+        (user_data_object->base.type == (vx_enum)VX_TYPE_USER_DATA_OBJECT))
+    {
+        /* assign reference type specific callback's */
+        user_data_object->base.destructor_callback =
```

Comment:
These 4 callbacks are assigned the same between this function and the createuserdata object function.  Just for maintaining consistency in the code and avoiding duplication and potential deviation, perhaps they can be grouped into a single static function in this file that can be called from both of these functions.

Reply:
done